src/ts/components/forms/slider.tsx

```diff
@@ -0,0 +1,358 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { ComponentProps } from '../../types';
+import {
+  convertPercentageStepToStep,
+  convertStepToPercentageStep,
+  getClosestValue,
+  getStepSeries,
+} from '../../utils/slider';
+
+export const PERCENTAGE_MIN = 0;
+export const PERCENTAGE_MAX = 1;
+
+export type Range = Readonly<{
+  from: number;
+  to: number;
+}>;
+
+export interface SliderProps extends ComponentProps {
+  /**
+   * Minimum value allowed to be selected in range
+   */
+  min: number;
+  /**
+   * Minimum value allowed to be selected in range
+   */
+  max: number;
+  /**
+   * Stepsize for increasing/decreasing range
+   */
+  step: number;
+  /**
+   * Initial values for from and to value
+   */
+  initialValue?: Range;
+  /**
+   * What should happen to range value on change?
+   */
+  onChange: (value: Range) => void;
+}
+
+export interface MousePosition {
+  x: number;
+  y: number;
+}
+
+export interface StateProps {
+  from: number;
+  to: number;
+  mouseDown?: MousePosition;
+}
+
+export interface SyntheticMouseEvent {
+  type: string;
+  clientX: number;
+}
+
+export interface SyntheticTouchEvent {
+  type: string;
+  targetTouches: TouchList | React.TouchList;
+}
+
+const isSyntheticMouseEvent = (
+  event: SyntheticMouseEvent | SyntheticTouchEvent
+): event is SyntheticMouseEvent => event.hasOwnProperty('clientX');
+
+export class Slider extends React.PureComponent<SliderProps, StateProps> {
+  private unsubscribers: Array<() => void> = [];
+  private stepSeries: ReadonlyArray<number>;
+  private percentageStepSeries: ReadonlyArray<number>;
+  private onHandleMouseFrom: (event: SyntheticMouseEvent) => void;
+  private onHandleMouseTo: (event: SyntheticMouseEvent) => void;
+  private onHandleTouchFrom: (event: SyntheticTouchEvent) => void;
+  private onHandleTouchTo: (event: SyntheticTouchEvent) => void;
+
+  public constructor(props: SliderProps) {
+    super(props);
+
+    const { min, max, step } = props;
+
+    this.stepSeries = getStepSeries(step, min, max);
+    this.percentageStepSeries = this.stepSeries.map(value =>
+      convertStepToPercentageStep(min, max, value)
+    );
+
+    const initialValue = {
+      from: props.initialValue ? props.initialValue.from : min,
```

Comment:
This initial value stuff should probably to some smart checking if the min is larger than the max stuff. Same for the mix and max values themselves. Otherwise we may end up with a weird inverted slider.

Thinking about it... we may also have issues with negative numbers. What happens if the min is `-100` and the max is `0`?

Reply:
Agree we should check if min < max and probably also if initial values are within range 👍 

I checked and slider with negative values works absolutely fine.

Reply:
Don't think we can check if `min` is smaller than `max` using `TS` so have just made sure the `initialValues` are correct